include/xtensor/xpad.hpp

```diff
@@ -0,0 +1,68 @@
+/***************************************************************************
+* Copyright (c) 2016, Johan Mabille, Sylvain Corlay,  Wolf Vollprecht and  *
+* Martin Renou                                                             *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_PAD_HPP
+#define XTENSOR_PAD_HPP
+
+#include "xarray.hpp"
+#include "xtensor.hpp"
+#include "xview.hpp"
+#include "xstrided_view.hpp"
+
+namespace xt
+{
+    /**
+     * @brief Defines different algorithms to be used in ``xt::pad``
+     */
+    enum class pad_mode
+    {
+        periodic
+    };
+
+    template <class E, class S>
+    auto pad(E&& e, std::initializer_list<S> pad_width, pad_mode mode = pad_mode::periodic)
+    {
+        auto out = e;
+
+        for (std::size_t axis = 0; axis < e.shape().size(); ++axis)
+        {
+            auto n = pad_width.begin()[axis];
+
+            xt::xstrided_slice_vector sv_bgn;
+            xt::xstrided_slice_vector sv_end;
+
+            for (std::size_t a = 0; a < axis; ++a)
+            {
+                sv_bgn.push_back(xt::all());
+                sv_end.push_back(xt::all());
+            }
+
+            if (mode == pad_mode::periodic)
+            {
+                sv_bgn.push_back(xt::range(out.shape()[axis]-n, out.shape()[axis]));
+                sv_end.push_back(xt::range(0, n));
+            }
+
+            for (std::size_t a = axis+1; a < e.shape().size(); ++a)
+            {
+                sv_bgn.push_back(xt::all());
+                sv_end.push_back(xt::all());
+            }
+
+            auto bgn = xt::strided_view(out, sv_bgn);
+            auto end = xt::strided_view(out, sv_end);
```

Comment:
I think it would be better to build the `strided_view` on `e` instead of `out` (no need to forward `e` here, taking a reference is ok and will avoid dangling reference if `E` is an rvalue reference).

Reply:
I think the view should be `out`. The idea is that padding proceeds sequentially along all axes. For the example 

```cpp
{{0, 1, 2},
 {3, 4, 5}}
```

becomes 

```cpp
{{0, 1, 2},
 {3, 4, 5},
 {0, 1, 2},
 {3, 4, 5},
 {0, 1, 2},
 {3, 4, 5}}
```

and then

```cpp
{{0, 1, 2, 0, 1, 2, 0, 1, 2},
 {3, 4, 5, 3, 4, 5, 3, 4, 5},
 {0, 1, 2, 0, 1, 2, 0, 1, 2},
 {3, 4, 5, 3, 4, 5, 3, 4, 5},
 {0, 1, 2, 0, 1, 2, 0, 1, 2},
 {3, 4, 5, 3, 4, 5, 3, 4, 5}}
```

Taking the view on `e` would not allow this.

Reply:
ouch I missed the `out` is computed in the loop. Shame on me! Ignore my comment ;)